mm2src/coins/solana.rs

```diff
@@ -307,6 +317,14 @@ async fn withdraw_impl(coin: SolanaCoin, req: WithdrawRequest) -> WithdrawResult
     withdraw_base_coin_impl(coin, req).await
 }
 
+type SolTxFut = Box<dyn Future<Item = SolTransaction, Error = TransactionErr> + Send + 'static>;
+
+impl Transaction for SolTransaction {
+    fn tx_hex(&self) -> Vec<u8> { serialize(self).unwrap() }
+
+    fn tx_hash_as_bytes(&self) -> BytesJson { BytesJson(Vec::from(self.signatures.get(0).unwrap().as_ref())) }
```

Comment:
The below is a better way to do this
```suggestion
impl Transaction for SolTransaction {
    fn tx_hex(&self) -> Vec<u8> {
        serialize(self).unwrap_or_else(|e| {
            error!("Error serializing SolTransaction: {}", e);
            vec![]
        })
    }

    fn tx_hash_as_bytes(&self) -> BytesJson {
        let hash = match self.signatures.get(0) {
            Some(signature) => signature,
            None => {
                error!("No signature found in SolTransaction");
                return BytesJson(Vec::new());
            },
        };
        BytesJson(Vec::from(hash.as_ref()))
    }
}
```
otherwise if unwrap is used straight away it can crash the maker node here https://github.com/KomodoPlatform/komodo-defi-framework/blob/36441d551796c0a29b82e4c3d3de4e22417cbe98/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L750 by sending a transaction with no signature through the p2p network.